src/GafferScene/Deformer.cpp

```diff
@@ -0,0 +1,170 @@
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferScene;
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( Deformer );
+
+size_t Deformer::g_firstPlugIndex = 0;
+
+Deformer::Deformer( const std::string &name )
+	:	ObjectProcessor( name )
+{
+	init();
+}
+
+Deformer::Deformer( const std::string &name, size_t minInputs, size_t maxInputs )
+	:	ObjectProcessor( name, minInputs, maxInputs )
+{
+	init();
+}
+
+void Deformer::init()
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new BoolPlug( "adjustBounds", Plug::In, true ) );
+	// Remove pass-through created by base class
+	outPlug()->boundPlug()->setInput( nullptr );
+}
+
+Deformer::~Deformer()
+{
+}
+
+Gaffer::BoolPlug *Deformer::adjustBoundsPlug()
+{
+	return getChild<BoolPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::BoolPlug *Deformer::adjustBoundsPlug() const
+{
+	return getChild<BoolPlug>( g_firstPlugIndex );
+}
+
+void Deformer::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ObjectProcessor::affects( input, outputs );
+
+	if( input == outPlug()->objectPlug() || input == inPlug()->boundPlug() || input == adjustBoundsPlug() )
+	{
+		outputs.push_back( outPlug()->boundPlug() );
+	}
+}
+
+bool Deformer::adjustBounds() const
+{
+	return adjustBoundsPlug()->getValue();
+}
+
+void Deformer::hashBound( const ScenePath &path, const Gaffer::Context *context, const ScenePlug *parent, IECore::MurmurHash &h ) const
+{
+	if( adjustBounds() )
+	{
+		const PathMatcher::Result m = filterValue( context );
+		if( m & ( PathMatcher::ExactMatch | PathMatcher::DescendantMatch ) )
+		{
+			ObjectProcessor::hashBound( path, context, parent, h );
+			if( m & PathMatcher::ExactMatch )
+			{
+				outPlug()->objectPlug()->hash( h );
+			}
+			else
+			{
+				inPlug()->objectPlug()->hash( h );
+			}
+
+			if( m & PathMatcher::DescendantMatch )
+			{
+				h.append( hashOfTransformedChildBounds( path, outPlug() ) );
+			}
+			else
+			{
+				h.append( hashOfTransformedChildBounds( path, inPlug() ) );
+			}
+			return;
+		}
+		// Fall through to pass-through
+	}
+
+	h = inPlug()->boundPlug()->hash();
+}
+
+Imath::Box3f Deformer::computeBound( const ScenePath &path, const Gaffer::Context *context, const ScenePlug *parent ) const
+{
+	if( adjustBounds() )
+	{
+		const PathMatcher::Result m = filterValue( context );
+		if( m & ( PathMatcher::ExactMatch | PathMatcher::DescendantMatch ) )
+		{
+			// Need to compute new bounds. This consists of the bounds of
+			// the (potentially deformed) object at this location and the
+			// (potentially deformed) bounds of our children.
+			Box3f result;
+			if( m & PathMatcher::ExactMatch )
+			{
+				// Get bounds from deformed output object.
+				/// \todo Some derived classes may be able to compute an output bound
+				/// for the object without computing the full deformation. We could add
+				/// virtual methods to allow that.
+				result.extendBy( SceneAlgo::bound( outPlug()->objectPlug()->getValue().get() ) );
```

Comment:
I'm conscious that I've already pushed the scope of CopyPrimitiveVariables by including these larger-scale improvements, and am reluctant to push the scope further. I'll spend an hour or two now to explore what it might look like though, and report back...

Reply:
Any reports on this exploration?

Reply:
I haven't got anywhere with it I'm afraid. My preference would be to hold off rather than creep the scope any more.